src/path.jl

```diff
@@ -706,8 +706,8 @@ end
 
 Performs a depth first search through the directory structure
 """
-function walkpath(fp::AbstractPath; topdown=true, follow_symlinks=false, onerror=throw)
-    return Channel() do chnl
+function walkpath(fp::P; topdown=true, follow_symlinks=false, onerror=throw)  where P <: AbstractPath
+    return Channel(ctype=P) do chnl
```

Comment:
Nice!  Has `ctype` always been a thing I could use?

Reply:
Looks like it exited all the way back on 0.6...

Reply:
yes, though it is deprecated at some point.
the new one is `Channel{P}() do` but that that doesn't work in 1.0